tasks/managed/rh-sign-image/rh-sign-image.yaml

```diff
@@ -336,40 +326,83 @@ spec:
                     sourceTag=${tag}-source
 
                     if ! grep -q "^${registry_reference}:${sourceTag}$" "/tmp/${sourceContainerDigest}" ; then
-                      while (( ${RUNNING_JOBS@P} >= "$CONCURRENT_LIMIT" )); do
-                        wait -n
-                      done
-
-                      echo "Creating ${requestType} to sign image with tag ${tag}:"
-                      echo "- reference=${registry_reference}:${sourceTag}"
-                      echo "- manifest_digest=${sourceContainerDigest}"
-                      echo "- requester=$(params.requester)"
-
-                      ${requestType} --pipeline "${request}" \
-                        -p pipeline_image="${pipeline_image}" \
-                        -p reference="${registry_reference}:${sourceTag}" \
-                        -p manifest_digest="${sourceContainerDigest}" \
-                        -p requester="$(params.requester)" \
-                        -p config_map_name="${config_map_name}" \
-                        -p taskGitUrl="$(params.taskGitUrl)" \
-                        -p taskGitRevision="$(params.taskGitRevision)" \
-                        -l ${TASK_LABEL}="${TASK_ID}" \
-                        -l ${PIPELINERUN_LABEL}="$(params.pipelineRunUid)" \
-                        -t "$(params.requestTimeout)" --pipeline-timeout "0h30m0s" --task-timeout "0h25m0s" \
-                        "${EXTRA_ARGS[@]}" -s true &
+                      to_sign_references+=("${registry_reference}:${sourceTag}")
+                      to_sign_digests+=("${manifest_digest}")
                     else
                       echo "Signature already exists for:"
                       echo "- reference=${registry_reference}:${sourceTag}"
                       echo "- manifest_digest=${sourceContainerDigest}"
                     fi
-
-                  ((++REQUEST_COUNT))
-                  echo "Request Count: $REQUEST_COUNT"
                 done
               done
             fi
         done
 
+        references_batch=""
+        digests_batch=""
+
+        # Process to sign arrays in batches
+        for i in "${!to_sign_references[@]}"; do
+          new_references_batch="${references_batch}${to_sign_references[$i]} "
+          new_digests_batch="${digests_batch}${to_sign_digests[$i]} "
+
+          # if batches are too big, send the request
+          if [[ ${#new_references_batch} -gt $BATCH_LIMIT || ${#new_digests_batch} -gt $BATCH_LIMIT ]]; then
+
+            while (( ${RUNNING_JOBS@P} >= "$CONCURRENT_LIMIT" )); do
+              wait -n
+            done
+
+            echo "Creating ${requestType} to sign images: ${references_batch}"
+            echo "- digests: ${digests_batch[*]}"
+
+            ${requestType} \
```

Comment:
Are we sure this will work correctly via internal-pipeline? In theory it should, it doesn't rely on any new secrets, right? But I think it would be good to test this somehow - Scott Hebert can help with that.

Reply:
I'm not 100% sure, but I assume it should work, I don't see reason why it shouldn't